statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,622 @@
+            self.sclust = clust.copy()
+
+        # The number of clusters per stratum
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.clust_per_strat = np.bincount(self.strat[ii])
+
+        # The stratum for each cluster
+        self.strat_for_clust = self.strat[ii]
+
+        # The fpc for each cluster
+        self.fpc = self.fpc[ii]
+
+        # The total number of clusters over all stratum
+        self.nclust = np.sum(self.clust_per_strat)
+
+        # get indices of all clusters within a stratum
+        self.ii = []
+        for s in range(self.nstrat):
+            self.ii.append(np.flatnonzero(self.strat_for_clust == s))
+
+    def __str__(self):
+        """
+        The __str__ method for our data
+        """
+        summary_list = ["Number of observations: ", str(len(self.strat)),
+                        "Sum of weights: ", str(self.weights.sum()),
+                        "Number of strata: ", str(self.nstrat),
+                        "Number of clusters per stratum: ", str(self.clust_per_strat),
+                        "Method to compute SE: ", self.cov_method]
+
+        return "\n".join(summary_list)
+
+    def _check_args(self, strata, cluster, weights, fpc):
+        """
+        Minor error checking to make sure user supplied any of
+        strata, cluster, or weights. For unspecified subgroup labels
+        an array of ones is created
+
+        Parameters
+        ----------
+        strata : array-like or None
+            Strata for each observation. If none, an array
+            of ones is constructed
+        cluster : array-like or None
+            Cluster for each observation. If none, an array
+            of ones is constructed
+        weights : array-like or None
+            The weight for each observation. If none, an array
+            of ones is constructed
+
+        Returns
+        -------
+        vals[0] : ndarray
+            array of the strata labels
+        vals[1] : ndarray
+            array of the cluster labels
+        vals[2] : ndarray
+            array of the observation weights
+        """
+        if all([x is None for x in (strata, cluster, weights)]):
+            raise ValueError("""At least one of strata, cluster, rep_weights, and weights
+                             musts not be None""")
+        v = [len(x) for x in (strata, cluster, weights) if x is not None]
+        if len(set(v)) != 1:
+            raise ValueError("""lengths of strata, cluster, and weights
+                             are not compatible""")
+        n = v[0]
+        vals = []
+        for x in (strata, cluster, weights):
+            if x is None:
+                vals.append(np.ones(n))
+            else:
+                vals.append(np.asarray(x))
+
+        if fpc is None:
+            vals.append(np.zeros(n))
+        else:
+            vals.append(np.asarray(fpc))
+
+        return vals[0], vals[1], vals[2], vals[3]
+
+    def get_rep_weights(self, c=None, bsn=None):
```

Comment:
Josef suggest that we add cov_method as an argument here and not attach it to the class.  Let's go with that approach.